src/core/components/hv-root/index.tsx

```diff
@@ -220,7 +220,10 @@ export default class Hyperview extends PureComponent<HvScreenProps.Props> {
       if (navigation && doc) {
         const { behaviorElement, delay, newElement, targetId } = options;
         const delayVal: number = +(delay || '');
-        navigation.setUrl(options.onUpdateCallbacks.getState().url || '');
+        const state = options.onUpdateCallbacks.getState();
+        if (state.url) {
+          navigation.setUrl(state.url);
```

Comment:
we no longer set the URL to '' - there's no side effect with this change?

Reply:
The url in `navigation` is used to get the domain for other requests, it seems to have no other purpose. Presumably once it's set once, the domain should remain the same.

setUrl() only accepts `string` not `string | null`. The previous code I had in there was to work around state's url value supporting null values. I also changed the state setter in `hv-route` so that doc and url will always either both have a value or neither will have a value, so this part of the code should not actually ever be reached if the doc+url are null. (There is a null check for `doc` a little higher up)

Reply:
Code rewritten to improve intention